test/acceptance_with_go_client/helper.go

```diff
@@ -91,32 +91,41 @@ func GetVectors(t *testing.T,
 		require.True(t, ok)
 
 		for _, targetVector := range targetVectors {
-			switch vector := vectors[targetVector].(type) {
-			case []interface{}:
-				var multiVector [][]float32
-				var vec []float32
-				for i := range vector {
-					switch v := vector[i].(type) {
-					case float64:
-						vec = append(vec, float32(v))
-					case []interface{}:
-						multiVectorVector := make([]float32, len(v))
-						for j := range v {
-							multiVectorVector[j] = float32(v[j].(float64))
-						}
-						multiVector = append(multiVector, multiVectorVector)
-					}
-				}
-				if len(multiVector) > 0 {
-					targetVectorsMap[targetVector] = multiVector
-				} else {
-					targetVectorsMap[targetVector] = vec
-				}
-			default:
-				panic(fmt.Sprintf("unexpected vector types in GraphQL response: %T", vectors[targetVector]))
+			if targetVector == "" {
+				targetVectorsMap[""] = parseVector(additional["vector"])
+				continue
 			}
+
+			targetVectorsMap[targetVector] = parseVector(vectors[targetVector])
 		}
 	}
 
 	return targetVectorsMap
 }
+
+func parseVector(data interface{}) models.Vector {
+	vector, ok := data.([]interface{})
+	if !ok {
+		panic(fmt.Sprintf("unexpected vector types in GraphQL response: %T", data))
+	}
+
+	var multiVector [][]float32
+	var vec []float32
+	for i := range vector {
+		switch v := vector[i].(type) {
+		case float64:
+			vec = append(vec, float32(v))
+		case []interface{}:
```

Comment:
```suggestion
		case [][]float32:
```

I think you can safely expect `[][]float32`, since there's an unchecked conversion below anyways

Similarly to how it's done in other places:
https://github.com/weaviate/weaviate/blob/a443b9c3af1490607b3cee1ca2b756de7cee22dc/adapters/handlers/grpc/v1/parse_search_request.go#L1307

https://github.com/weaviate/weaviate/blob/a443b9c3af1490607b3cee1ca2b756de7cee22dc/usecases/vectorizer/combine.go#L19

Reply:
This would not work as every number in the array is wrapped in an interface, so you need to cast them all. Probably those other places have some other serde path

![image](https://github.com/user-attachments/assets/324e385e-3bba-4f06-8294-1b655793e8af)